### ISSUE TITLE
`unnecessary_find_map` should have `Applicability::HasPlaceholders`?

### ISSUE THREAD

**Issue by ada4a** [rust-lang/rust-clippy#14486]
### Summary

~It seems like when the original code is long enough,~ (EDIT: nope, it happens even with short original code, realized when minimizing the reproducer) The suggested replacement is shortened using ellipses. But I still get a code action to apply the suggestion, which would result in bogus code.

I'm using LazyVim with rustaceanvim v5.25.2 if that's relevant

### Reproducer

I tried this code:

```rust
let _ = [1].iter().find_map(|_| Some(()));
```

I expected to see this happen:
one of two things:
- I get a warning without a suggestion
- I get a warning with the correct suggestion:
```rust
let _ = [1].iter().map(|_| ()).next();
```

Instead, this happened:
I got a warning with the following suggestion: 
```
Diagnostics:
1. this `.find_map` can be written more simply
   for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_find_map [unnecessary_find_map]
2. try instead: `map(..).next()` [unnecessary_find_map]
```

### Version

```text
rustc 1.85.0 (4d91de4e4 2025-02-17) (built from a source tarball)
binary: rustc
commit-hash: 4d91de4e48198da2e33413efdcd9cd2cc0c46688
commit-date: 2025-02-17
host: x86_64-unknown-linux-gnu
release: 1.85.0
LLVM version: 19.1.7
```

**Comment by ada4a** [comment]
Wow, thanks for such a fast fix!:)
